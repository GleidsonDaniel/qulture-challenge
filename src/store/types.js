export const INITIAL_STATE = 'INITIAL_STATE';

// user reducer
export const SET_USER_LIST_USER_REDUCER = 'SET_USER_LIST_USER_REDUCER';
export const SET_SELECTED_USER_USER_REDUCER = 'SET_SELECTED_USER_USER_REDUCER';
export const SET_FETCHING_USERS_USER_REDUCER =
  'SET_FETCHING_USERS_USER_REDUCER';
