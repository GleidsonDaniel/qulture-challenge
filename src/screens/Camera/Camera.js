import React, {Component} from 'react';
import {Text, TouchableOpacity, View, ImageBackground} from 'react-native';
import {RNCamera} from 'react-native-camera';
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Camera extends Component {
  constructor(props) {
    super(props);
    this.state = {
      image: null,
    };
  }

  takePicture = async () => {
    if (this.camera) {
      const options = {quality: 0.5, base64: true};
      const data = await this.camera.takePictureAsync(options);
      this.setState({image: data});
    }
  };

  render() {
    const {image} = this.state;
    const {onPress} = this.props;
    return image && image.uri ? (
      <ImageBackground style={{flex: 1}} source={{uri: image.uri}}>
        <View style={{flex: 8}} />
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'space-around',
            padding: 5,
            flex: 1,
          }}>
          <TouchableOpacity onPress={() => this.setState({image: null})}>
            <Icon name="times" size={25} color="#fff" />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => onPress(image)}>
            <Icon name="check" size={25} color="#fff" />
          </TouchableOpacity>
        </View>
      </ImageBackground>
    ) : (
      <View
        style={{flex: 1, flexDirection: 'column', backgroundColor: 'black'}}>
        <RNCamera
          ref={ref => {
            this.camera = ref;
          }}
          style={{flex: 1, justifyContent: 'flex-end', alignItems: 'center'}}
          type={RNCamera.Constants.Type.back}
          flashMode={RNCamera.Constants.FlashMode.on}
          androidCameraPermissionOptions={{
            title: 'Permission to use camera',
            message: 'We need your permission to use your camera',
            buttonPositive: 'Ok',
            buttonNegative: 'Cancel',
          }}
          androidRecordAudioPermissionOptions={{
            title: 'Permission to use audio recording',
            message: 'We need your permission to use your audio',
            buttonPositive: 'Ok',
            buttonNegative: 'Cancel',
          }}
        />
        <View style={{flex: 0, flexDirection: 'row', justifyContent: 'center'}}>
          <TouchableOpacity
            onPress={() => this.takePicture()}
            style={{
              flex: 0,
              backgroundColor: '#fff',
              borderRadius: 5,
              padding: 15,
              paddingHorizontal: 20,
              alignSelf: 'center',
              margin: 20,
            }}>
            <Text style={{fontSize: 14}}> SNAP </Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
