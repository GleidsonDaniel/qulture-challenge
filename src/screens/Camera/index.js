import React from 'react';
import {useDispatch} from 'react-redux';
import {useNavigation} from 'react-navigation-hooks';

import {setPhoto} from '~/actions/cameraActions';

import CameraComponent from './Camera';

export default function Camera() {
  const dispatch = useDispatch();
  const {goBack} = useNavigation();

  const handleImage = image => {
    dispatch(setPhoto(image));
    goBack();
  };

  return <CameraComponent onPress={image => handleImage(image)} />;
}
