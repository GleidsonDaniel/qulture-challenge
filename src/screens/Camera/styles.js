import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  flex-direction: column;
  background-color: #000;
`;

const TakeShotContainer = styled.View`
  flex: 0;
  flex-direction: row;
  justify-content: center;
`;

const TakeShotButton = styled.TouchableOpacity`
  flex: 0;
  background-color: #fff;
  border-radius: 5px;
  padding: 15px;
  align-self: center;
  margin: 20px;
`;

export {Container, TakeShotContainer, TakeShotButton};
