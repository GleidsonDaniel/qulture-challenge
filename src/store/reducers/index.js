import {combineReducers} from 'redux';

import InitialReducer from './InitialReducer';

const Reducers = combineReducers({InitialReducer});

export default Reducers;
