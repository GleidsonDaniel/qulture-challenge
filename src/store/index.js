import thunk from 'redux-thunk';
import multi from 'redux-multi';
import {createStore, applyMiddleware, compose} from 'redux';
import {persistStore, persistReducer} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import reducers from './reducers';
import tron from '~/config/ReactotronConfig';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);
const middleware = applyMiddleware(multi, thunk);
const reactotron = __DEV__ && tron.createEnhancer();
const combine = __DEV__
  ? compose(
      middleware,
      reactotron,
    )
  : compose(middleware);

const finalStore = createStore(persistedReducer, combine);

export const store = finalStore;
export const persistor = persistStore(finalStore);
