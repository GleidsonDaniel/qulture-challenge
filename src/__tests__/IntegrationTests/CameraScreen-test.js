import React from 'react';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';

import Camera from '~/screens/Camera';
import {store} from '~/store';

describe('Camera', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <Provider store={store}>
          <Camera />
        </Provider>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
