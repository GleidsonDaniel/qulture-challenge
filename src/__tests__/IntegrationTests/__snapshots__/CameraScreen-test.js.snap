// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
